numpyro/distributions/distribution.py

```diff
@@ -350,6 +350,45 @@ def variance(self):
         return np.broadcast_to(self.base_dist.variance, self.batch_shape + self.event_shape)
 
 
+class ImproperUniform(Distribution):
+    """
+    A helper distribution with zero :meth:`log_prob` and improper
+    :meth:`sample`.
+
+    .. note:: `sample` method does not generate samples distributed uniformly over the support.
+        Instead, it only returns valid values belong to the support.
+    """
+    arg_constraints = {}
+    unconstrained_radius = 2
+
+    def __init__(self, support, event_shape, batch_shape=(), validate_args=None):
+        self.support = support
+        super().__init__(batch_shape, event_shape, validate_args=validate_args)
+
+    def sample(self, key, sample_shape=()):
+        transform = biject_to(self.support)
+        prototype_value = np.zeros(self.event_shape)
+        unconstrained_event_shape = np.shape(transform.inv(prototype_value))
+        shape = sample_shape + self.batch_shape + unconstrained_event_shape
+        unconstrained_samples = random.uniform(key, shape,
+                                               minval=-self.unconstrained_radius,
+                                               maxval=self.unconstrained_radius)
+        return transform(unconstrained_samples)
```

Comment:
Do you plan to replace this with `NotImplementedError` and require uses to use `InitMessenger` or similar (which is more flexible)?

Reply:
I plan to do so. I need to change something in how to interpret shape of initial unconstrained parameters (which works for improper distribution but might not work for general priors).